### PR TITLE
Replace `base64` by `data-encoding` in libparsec_client_connection

### PR DESCRIPTION
**Pull request by touilleMan**

### CODE REVIEW

**Comment by FirelightFlagboy** [comment]
Is it related to https://mastodon.social/@rust_discussions/111177778310492873 ?

If that's the case can you make the comment more clear about it

**Comment by touilleMan** [comment]
> Is it related to https://mastodon.social/@rust_discussions/111177778310492873 ?

Partially: since we already use `data-encoding` for base32, I figured out we might as well use it for base64 to have a ergonomic and unified API on this encoding stuff

(okay I considered myself personally insulted by the quote "Rust is not the language for you if you don't like traits" :trollface: )

File: Cargo.toml
Comment (FirelightFlagboy) [inline Cargo.toml:1]: We still has `base64` & `base32` listed as dependencies

Comment (touilleMan) [reply Cargo.toml:1]: Good catch !

`base32` was still used by `libparsec_crypto`, I've fixed that (it is especially good to remove `base32` crate as it hasn't had update in 5years ^^)

`base64` is required by 3rd party libraries so there not much we can do here

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**